src/wui/seafaring_statistics_menu.cc

```diff
@@ -210,16 +210,16 @@ SeafaringStatisticsMenu::SeafaringStatisticsMenu(InteractivePlayer& plr,
 			   switch (note.action) {
 			   case Widelands::NoteShip::Action::kDestinationChanged:
 			   case Widelands::NoteShip::Action::kWaitingForCommand:
+			   case Widelands::NoteShip::Action::kNoPortLeft:
 			   case Widelands::NoteShip::Action::kGained:
 				   assert(note.ship != nullptr);
 				   update_ship(*note.ship);
-				   break;
+				   return;
 			   case Widelands::NoteShip::Action::kLost:
 				   remove_ship(note.ship->serial());
-				   break;
```

Comment:
this line should break 100% the same way, lets try :-)

Reply:
I'm not sure what you mean here?

In master this `switch` has a `default` clause which calls `NEVER_HERE()`. But this is incorrect because the `kNoPortLeft` action would fall there, causing the bug. I changed this to expkcitly include that action in the `switch` statement, and since my compiler complains about a `default` clause in a `switch` that covers all enum constants, I have all `switch` entries call `return` and moved the `NEVER_HERE()` below the `switch` instead.

---

gdb tells me that `kLost` is triggered correctly after sinking the ship. It occurs not directly after clicking the sink button but several seconds later when the sinking animation is over and the ship gets actually removed

Reply:
Hmm, the break: should directly run into the  NEVER_HERE(); ?
I dowlanded the built version and indeed it fxes the crash.